press/marketplace/doctype/marketplace_app_payment/marketplace_app_payment.py

```diff
@@ -6,17 +6,52 @@
 
 
 class MarketplaceAppPayment(Document):
-	def has_threshold_passed(self):
-		exchange_rate = frappe.db.get_single_value("Press Settings", "usd_rate")
-		total = self.total_usd + (self.total_inr / exchange_rate) if exchange_rate > 0 else 80
-		return total >= frappe.db.get_single_value("Press Settings", "threshold")
-
-	def get_commission(self, total):
-		# TODO: Handle partial commission
-		# if first month collection: $20, second month: $1000 and $500 - cap/threshold
-		# then commission should be calculated for $520 from second month collection onwards
-		return (
-			total * frappe.db.get_single_value("Press Settings", "commission")
-			if self.has_threshold_passed()
-			else total
-		)
+	# begin: auto-generated types
+	# This code is auto-generated. Do not modify anything in this block.
+
+	from typing import TYPE_CHECKING
+
+	if TYPE_CHECKING:
+		from frappe.types import DF
+
+		app: DF.Link | None
+		commission_inr: DF.Currency
+		commission_usd: DF.Currency
+		team: DF.Link | None
+		total_inr: DF.Currency
+		total_usd: DF.Currency
+	# end: auto-generated types
+
+	def get_commission(self, row_amount, currency):
+		"""
+		Calculate commission based on threshold logic:
+		- Below threshold: Commission = min(row_amount, remaining_to_reach_threshold)
+		- At/above threshold: Commission = row_amount * commission_rate
+		"""
+		press_settings = frappe.get_cached_doc("Press Settings")
+		exchange_rate = press_settings.usd_rate or 80
+		threshold = press_settings.threshold or 500
+		commission_rate = press_settings.commission or 0.20
+
+		# Calculate current total in USD
+		current_total = self.total_usd + (self.total_inr / exchange_rate)
+
+		if current_total >= threshold:
```

Comment:
```suggestion
		if current_total > threshold:
```
We can only check for threshold > total and not equals to since it will be handled later